applications/FluidDynamicsBiomedicalApplication/python_scripts/apply_parabolic_inlet_process.py

```diff
@@ -0,0 +1,117 @@
+    @classmethod
+    def GetDefaultParameters(cls):
+        default_settings = KratosMultiphysics.Parameters("""{
+            "wall_model_part_name": "",
+            "inlet_model_part_name": "",
+            "parabola_vertex_value" : 0.0,
+            "interval" : [0.0,"End"],
+            "local_axes" : {},
+            "parallel_distance_max_levels" : 25
+        }""")
+        return default_settings
+
+    def ExecuteInitialize(self):
+        # Get and check domain size
+        inlet_model_part = self.model.GetModelPart(self.settings["inlet_model_part_name"].GetString())
+        domain_size = inlet_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        if domain_size not in [2,3]:
+            raise ValueError(f"Wrong 'DOMAIN_SIZE' value {domain_size} in ProcessInfo container.")
+
+        # Set the INLET flag in the inlet model part nodes and conditions
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.INLET, True, inlet_model_part.Nodes)
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.INLET, True, inlet_model_part.Conditions)
+
+        # Compute the normal on the nodes of interest
+        # Note that a custom normal variable is use to avoid interfering with the wall one
+        KratosMultiphysics.NormalCalculationUtils().CalculateOnSimplexNonHistorical(
+            inlet_model_part,
+            domain_size,
+            KratosBio.INLET_NORMAL)
+
+        # Prepare skin for wall distance calculation
+        max_levels = self.settings["parallel_distance_max_levels"].GetInt()
+        wall_model_part = self.model.GetModelPart(self.settings["wall_model_part_name"].GetString())
+        root_model_part = wall_model_part.GetRootModelPart()
+        KratosBio.ParabolicProfileUtilities.CalculateWallParallelDistance(wall_model_part, root_model_part, max_levels)
+
+    def ExecuteBeforeSolutionLoop(self):
+        self.ExecuteInitializeSolutionStep()
+
+    def ExecuteInitializeSolutionStep(self):
+        inlet_model_part = self.model.GetModelPart(self.settings["inlet_model_part_name"].GetString())
+        current_time = inlet_model_part.ProcessInfo[KratosMultiphysics.TIME]
+        if self.interval.IsInInterval(current_time):
+            self.step_is_active = True
+            if self.max_value_is_numeric:
+                KratosBio.ParabolicProfileUtilities.ImposeParabolicInlet(inlet_model_part, self.max_value)
+            else:
+                KratosBio.ParabolicProfileUtilities.ImposeParabolicInlet(inlet_model_part, self.max_value_function)
+
+    def ExecuteFinalizeSolutionStep(self):
+        # Here we free all of the nodes in the inlet
+        if self.step_is_active:
+            inlet_model_part = self.model.GetModelPart(self.settings["inlet_model_part_name"].GetString())
+            KratosBio.ParabolicProfileUtilities.FreeParabolicInlet(inlet_model_part)
+        self.step_is_active = False
+
+    @classmethod
+    def _ReturnDistanceProcessPrototype(cls, domain_size):
```

Comment:
Why `classmethod` and not `staticmethod`?

Reply:
Changing it.